### PR TITLE
NLog ExceptionlessTarget SetDefaultMinLogLevel

### PR DESCRIPTION
**Pull request by snakefoot**
See also: https://github.com/exceptionless/Exceptionless.Net/issues/260

### CODE REVIEW

File: src/Platforms/Exceptionless.NLog/ExceptionlessTarget.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ public class ExceptionlessTarget : TargetWithLayout {
                         config.ServerUrl = serverUrl;
                     config.UseLogger(new NLogInternalLoggger());
                     config.UseInMemoryStorage();
+                    config.SetDefaultMinLogLevel(Logging.LogLevel.Trace);   // Rely on NLog Logging Rules
```
Comment (niemyjski) [inline src/Platforms/Exceptionless.NLog/ExceptionlessTarget.cs:42]: This works, but I wonder if there is a way to see what the min log level is for a given target? @ejsmith I kinda feel like maybe if you specify a logging target like nlog which doesn't have a restrict on the target itself, than this is the correct approach.

Comment (snakefoot) [reply src/Platforms/Exceptionless.NLog/ExceptionlessTarget.cs:42]: NLog Logger-output will never reach a NLog Target without first specifying a NLog Logging-Rule that redirects output from matching NLog Logger-names / LogLevels. See also https://github.com/NLog/NLog/wiki/Configuration-file#rules

**Review by niemyjski — APPROVED** [review]
Thanks for the PR!

**Comment by snakefoot** [comment]
Maybe the override-logic of the call to `SetDefaultMinLogLevel` on the default-client should be: "override if not configured already" ?

**Comment by niemyjski** [comment]
Yeah we could add an extension or check for that, wouldn't be a bad idea.

**Comment by niemyjski** [comment]
@snakefoot I just looked at the implementation and it already handles this :)

```cs
        public void SetDefaultMinLogLevel(LogLevel minLogLevel) {
            if (!Settings.ContainsKey("@@log:*"))
                Settings.Add("@@log:*", minLogLevel.ToString());
        }
```

**Pull request merged**